src/components/znkSession/services/znkSession.service.js

```diff
@@ -0,0 +1,203 @@
+                        sessionGUID: UtilitySrv.general.createGuid(),
+                        educatorUID: userAuth.uid || 'N/A',
+                        studentUID: StudentContextSrv.getCurrUid() || 'N/A',
+                        extendTime: 0,
+                        startTime: Date.now(),
+                        duration: null,
+                        sessionSubject: sessionSubject.id,
+                        status: SessionsStatusEnum.ACTIVE.enum  //(values: 1 = Active, 0 = Ended)
+                    };
+                }
+                function getKeyByValue(obj, value) {
+                    for( var prop in obj ) {
+                        if( obj.hasOwnProperty( prop ) ) {
+                            if( obj[ prop ] === value ) {
+                                return prop;
+                            }
+                        }
+                    }
+                }
+                function minToUnixTimestamp(min) {
+                    return min * 60 * 1000;
+                }
+                function getPath(param) {
+                    if (!userAuth) {
+                        $log.error('Invalid user');
+                        return;
+                    }
+                    var path;
+                    var educatorUID = isTeacherApp ? userAuth.uid : TeacherContextSrv.getCurrUid();
+                    var studentUID = isTeacherApp ? StudentContextSrv.getCurrUid() : userAuth.uid;
+                    switch (param) {
+                        case 'sessions':
+                            path = ENV.studentAppName + '/liveSession/' + currLiveSessionsGUID.guid;
+                            return path;
+                        case 'student':
+                            path = ENV.studentAppName + '/users/$$uid/liveSession';
+                            return path.replace('$$uid', '' + studentUID);
+                        case 'educator':
+                            path = ENV.dashboardAppName + '/users/$$uid/liveSession';
+                            return path.replace('$$uid', '' + educatorUID);
+                        default:
+                            return;
+                    }
+                }
+                function saveSession() {
+                    $log.debug('saveSession, sessionData: ', sessionData);
+                    var dataToSave = {};
+                    globalStorageProm.then(function (globalStorage) {
+                        var studentPath = getPath('student') + '/active';
+                        var educatorPath = getPath('educator') + '/active';
+                        var sessionPath = getPath('sessions');
+                        dataToSave[sessionPath] = sessionData;
+                        dataToSave[studentPath] = { guid: sessionData.sessionGUID };
+                        dataToSave[educatorPath] = { guid: sessionData.sessionGUID };
+                        globalStorage.update(dataToSave);
```

Comment:
doesn't it return promise (update)/

What will happen in case of error ? catch ?